### PR TITLE
Allow cosmetic filtering to also be applied to 1p items

### PR DESCRIPTION
**Pull request by antonok-edm**
Resolves https://github.com/brave/brave-browser/issues/8475

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

- Open brave://settings/shields and confirm that the first dropdown, labeled 'Block trackers & ads', is set to 'Trackers & ads blocked (standard)'.
- For each of the three possible values of the dropdown, visit http://raymondhill.net/ublock/tiles1.html and confirm the two colored tiles at the top of the page display their corresponding colors from the table below. Note that the page will need to be refreshed after each change of the setting. After observing each pair of colors, verify that the first dropdown in the 'Advanced View' of the Shields panel remains consistent with the selected setting.

| Block trackers & ads setting    | Left tile | Right tile |
|---------------------------------------|----------|-----------|
| Trackers & ads blocked (standard) | Green   | May display green for a moment while the page loads, but should remain red afterwards |
| Trackers & ads blocked (aggressive) | Green | Green |
| Allow all trackers & ads  | Red | Red |

- Confirm the behavior in the table above once again, this time changing the setting using the dropdown in the 'Advanced View' of the Shields panel. The page should refresh automatically when the setting is modified. Confirm that the setting in brave://settings/shields is *not* updated by changes to the Shields panel.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by pes10k** [comment]
@antonok-edm can we change the naming here to not be "uBlock mode", and something more descriptive of the functionality (1p cosmetic filtering, or global cosmetic filtering or general)?

**Comment by antonok-edm** [comment]
@pes - is this better?

**Comment by pes10k** [comment]
It would be good to add a second test to make sure this is not hidden when 1p cosmetic filtering is not enabled.  Otherwise, LGTM!

**Comment by antonok-edm** [comment]
@pes10k the old test right above the new one should already be handling that behavior

**Comment by pes10k** [comment]
Ah gotcha, then yep, LGTM!

**Comment by pes10k** [comment]
@antonok-edm just as one more quick ask, can you document here, in a paragraph or less, how QA can manually test?

**Comment by antonok-edm** [comment]
@pes10k is what's currently under the `Test Plan` header above sufficient?

**Comment by pes10k** [comment]
@antonok-edm ideally there'd be a fixed page, w/ a fixed result, we could point people too.  If we just need a place to host it, let me know and i can throw it up on my site

**Comment by antonok-edm** [comment]
I've rebased this to include #5521 so brave://flags works now.

Is http://raymondhill.net/ublock/tiles1.html okay as a test page? The right box appears red when the flag is disabled (or default) and green when enabled.

**Comment by pes10k** [comment]
@antonok-edm looks good to me :)

**Comment by antonok-edm** [comment]
@bridiver could I get your review?

**Comment by pes10k** [comment]
@rebron this is the issue discussed, ready for front end work

**Comment by karenkliu** [comment]
Designs added to https://github.com/brave/brave-browser/issues/8475

File: components/brave_shields/browser/ad_block_service_browsertest.cc [hunk 1]
```diff
@@ -799,7 +800,47 @@ class CosmeticFilteringDisabledTest : public AdBlockServiceTest {
...
+      "##.ad");
+
+  WaitForBraveExtensionShieldsDataReady();
+
+  GURL tab_url = embedded_test_server()->GetURL("b.com",
+                                                "/cosmetic_filtering.html");
+  ui_test_utils::NavigateToURL(browser(), tab_url);
+
+  content::WebContents* contents =
+    browser()->tab_strip_model()->GetActiveWebContents();
```
Comment (bridiver) [inline components/brave_shields/browser/ad_block_service_browsertest.cc:816]: lint nit: continuation of a line should be 4 spaces

File: components/brave_extension/extension/brave_extension/types/state/shieldsPannelState.ts [hunk 2]
```diff
@@ -8,7 +8,8 @@ import { NoScriptInfo } from '../other/noScriptInfo'
 import { SettingsData } from '../other/settingsTypes'
 
 export interface Tab {
-  cosmeticBlocking: boolean
+  cosmeticFiltering: BlockFPOptions
+  cosmeticFilteringFlag: boolean
```
Comment (bridiver) [inline components/brave_extension/extension/brave_extension/types/state/shieldsPannelState.ts:12]: why do we need a flag if we already have the setting?

File: browser/extensions/api/brave_shields_api.cc [hunk 3]
```diff
@@ -221,6 +221,50 @@ BraveShieldsGetCosmeticFilteringEnabledFunction::Run() {
...
 
+ExtensionFunction::ResponseAction
+BraveShieldsSetCosmeticFilteringControlTypeFunction::Run() {
+  std::unique_ptr<brave_shields::SetCosmeticFilteringControlType::Params>
+    params(
+      brave_shields::SetCosmeticFilteringControlType::Params::Create(*args_));
+  EXTENSION_FUNCTION_VALIDATE(params.get());
+
+  const GURL url(params->url);
+  // we don't allow setting defaults from the extension
```
Comment (bridiver) [inline browser/extensions/api/brave_shields_api.cc:232]: is there a test for this?

File: browser/resources/settings/default_brave_shields_page/default_brave_shields_page.js [hunk 4]
```diff
@@ -76,7 +92,10 @@ Polymer({
   },
 
   onAdControlChange_: function() {
```
Comment (bridiver) [inline browser/resources/settings/default_brave_shields_page/default_brave_shields_page.js:94]: this method could use a comment explaining what is happening here

File: components/brave_extension/extension/brave_extension/types/state/shieldsPannelState.ts (hunk 2 above)
Comment (antonok-edm) [reply components/brave_extension/extension/brave_extension/types/state/shieldsPannelState.ts:12]: I believe the idea is to add a way to decouple cosmetic filtering from network filtering, even though for general use cases they'll be switched together. cc @pes10k

File: browser/extensions/api/brave_shields_api.cc (hunk 3 above)
Comment (antonok-edm) [reply browser/extensions/api/brave_shields_api.cc:232]: I didn't specifically add one, but this uses the same logic as the other `set...ControlType` functions.

Any pointers on how I might test it? It seems like all the API calls are mocked out in the JS test suites.

Comment (bridiver) [reply browser/extensions/api/brave_shields_api.cc:232]: so no tests use the actual extension apis? That's a bit concerning, but I guess not something that needs to be dealt with in this PR

File: components/brave_extension/extension/brave_extension/types/state/shieldsPannelState.ts (hunk 2 above)
Comment (pes10k) [reply components/brave_extension/extension/brave_extension/types/state/shieldsPannelState.ts:12]: Hmm, im not sure I follow the above. We do want to be able to control cosmetic filtering independent of network filtering, but I don't follow why there are two different `cosmeticFiltering` controls above

Comment (antonok-edm) [reply components/brave_extension/extension/brave_extension/types/state/shieldsPannelState.ts:12]: @pes10k `cosmeticFilteringFlag` tracks the state of the feature flag. `cosmeticFiltering` tracks whether cosmetic filtering is for third-party only, everything, or completely off when the flag is on. Will rename/document to be more clear after discussion with @bridiver.

**Comment by antonok-edm** [comment]
Travis appeared to be disabled when this was merged, and some of the unit tests weren't fully updated - these are fixed in a followup PR https://github.com/brave/brave-core/pull/5657

**Pull request merged**